types/aws-lambda/index.d.ts

```diff
@@ -878,11 +880,17 @@ export interface SQSRecordAttributes {
     SenderId: string;
     ApproximateFirstReceiveTimestamp: string;
 }
+
+export type SQSMessageAttributeDataType = 'String' | 'Number' | 'Binary' | string;
+
 export interface SQSMessageAttribute {
-    Name: string;
-    Type: string;
-    Value: string;
+    stringValue?: string;
+    binaryValue?: string | Buffer;
```

Comment:
Can it really be either? I meant that I didn't know which :)

Reply:
[From here](https://github.com/aws/aws-sdk-js/blob/5dbd6fb9050035c1aaa4d952893f3d7941575d16/clients/sns.d.ts#L578), it's `Binary`, which is [defined here](https://github.com/aws/aws-sdk-js/blob/5dbd6fb9050035c1aaa4d952893f3d7941575d16/clients/sns.d.ts#L275) as `Buffer|Uint8Array|Blob|string`. `Uint8Array` is what `Buffer` extends and `Blob` is [defined here](https://github.com/aws/aws-sdk-js/blob/5dbd6fb9050035c1aaa4d952893f3d7941575d16/clients/sns.d.ts#L7) as just `{}`. Preference on what we go with?